crates/script/src/simulate.rs

```diff
@@ -112,10 +112,16 @@ impl PreSimulationState {
         // Executes all transactions from the different forks concurrently.
         let futs = transactions
             .into_iter()
-            .map(|transaction| async {
+            .map(|mut transaction| async {
                 let mut runner = runners.get(&transaction.rpc).expect("invalid rpc url").write();
+                let tx = transaction.tx_mut();
+
+                if let Some(tx) = tx.as_unsigned_mut() {
+                    if let Some(authorization_list) = &tx.authorization_list {
+                        runner.executor.set_delegation(authorization_list)?;
+                    }
```

Comment:
let's add a helper here to get the `authorization_list` https://github.com/foundry-rs/foundry/blob/c4a31a624874ab36284fca4e48d2197e43a62fbe/crates/common/src/transactions.rs#L223

signed transactions might have one as well

Reply:
Fixed, thanks